### ISSUE TITLE
Support Svelte 4

### ISSUE THREAD

**Issue by silenceofthewans** [rgossiaux/svelte-headlessui#152]
Changelog:
[https://github.com/sveltejs/svelte/blob/master/packages/svelte/CHANGELOG.md](https://github.com/sveltejs/svelte/blob/master/packages/svelte/CHANGELOG.md)

**Comment by rgossiaux** [comment]
Yes, I'll be hoping to release official Svelte 4 compatibility soon, but I haven't tested it out much yet because some of my dependencies need to be upgraded. 

In the meantime, please let me know if anyone has been using the library with Svelte 4 successfully or unsuccessfully. From what I've seen it should be fine, but this library's internals do some pretty sophisticated things in a few places, which is why I haven't just bumped the `peerDependencies` without doing some testing first.

**Comment by silenceofthewans** [comment]
I've been using `Menu` and `Listbox` in my app and they seem to be working fine can't say much for other components.

**Comment by Robin-w151** [comment]
I've been using the `Popover` component and everything is running smooth so far.

**Comment by AyeRlS** [comment]
It seems to work well, no issue so far

**Comment by MaximumMaxxx** [comment]
`Switch` componenet works fine

**Comment by vfrunza** [comment]
Listbox has been working fine for me too.

**Comment by MaximumMaxxx** [comment]
Has anyone found a workaround for the peer dependencies not being aligned to use while this issue sits in limbo?

**Comment by rgossiaux** [comment]
@MaximumMaxxx You should be able to just ignore the warning. I've been traveling lately but I plan on releasing an update to bump the peerDependencies in a week or two.

**Comment by padidas** [comment]
In case anyone is wondering how to overwrite the peer dependencies - this worked for me:  

  ```json
"devDependencies": {
    ...
    "@rgossiaux/svelte-headlessui": "^2.0.0",
    "svelte": "^4.1.2",
    ...
  },
  "overrides": {
    "@rgossiaux/svelte-headlessui": {
      "svelte": "$svelte"
    }
  }
```

**Comment by zach-hopkins** [comment]
> In case anyone is wondering how to overwrite the peer dependencies - this worked for me:  
> 
>   ```json
> "devDependencies": {
>     ...
>     "@rgossiaux/svelte-headlessui": "^2.0.0",
>     "svelte": "^4.1.2",
>     ...
>   },
>   "overrides": {
>     "@rgossiaux/svelte-headlessui": {
>       "svelte": "$svelte"
>     }
>   }
> ```

Thank you for this.

Have been using in production for two weeks with no issues @rgossiaux

**Comment by bradyisom** [comment]
I'm using with Svelte v4 and it looks like the `open` prop in `Disclosure` is not working.

**Comment by adminy** [comment]
Svelte 5 support? Unfortunately it's not backwards compatible and it's not as simple as version bump. Any plans to support svelte 5?

**Comment by Loizzus** [comment]
I have created a new fork for us to share so we can fix these issues: https://github.com/PyongyangOpenSource/svelte-headlessui
